omod/src/main/webapp/resources/scripts/conditionlist/controllers/condition.controller.js

```diff
@@ -153,27 +154,35 @@
             }
 
             self.saveCondition();
-        }
+        };
 
         self.getSelectedDate = self.getSelectedDate || function () {
-            return startDateElement.value;
-        }
+            return startDateElement.value ? self.toUTCDate(startDateElement.value) : null;
+        };
 
         self.getEndDate = self.getEndDate || function () {
-            return endDateElement.value;
-        }
+            return endDateElement.value ? self.toUTCDate(endDateElement.value) : null;
+        };
+
+        self.toUTCDate = function(dateString) {
+            var utcDate = null;
+            if (dateString) {
+                utcDate = new Date(dateString);
+                utcDate.setTime(utcDate.getTime() + utcDate.getTimezoneOffset() * 60 * 1000);
+            }
+            return utcDate;
+        };
 
```

Comment:
The above is what fixes the issue in RA-1678 with an inconsistent date getting stored in the database from what was entered by the user.  Here, I adapted a solution I found that was undertaken to fix a similar issue in the program enrollment components of coreapps.  Specifically, I was inspired by this commit:  https://github.com/openmrs/openmrs-module-coreapps/commit/50b1c92e929fbcdeed93e0748d0ca7e0521b7826

Reply:
Not sure about this, @mseaton , will comment on the ticket.